compiler/compile/OMRSymbolReferenceTable.cpp

```diff
@@ -2093,3 +2094,82 @@ OMR::SymbolReferenceTable::rememberOriginalUnimprovedSymRef(
       entryPreventingInsertion->second,
       original->getReferenceNumber());
    }
+
+
+const char *OMR::SymbolReferenceTable::_commonNonHelperSymbolNames[] =
+   {
+   "<contiguousArraySize>",
+   "<discontiguousArraySize>",
+   "<arrayClassRomPtr>",
+   "<javaLangClassFromClass>",
+   "<classFromJavaLangClass>",
+   "<addressOfClassOfMethod>",
+   "<componentClass>",
+   "<isArray>",
+   "<isClassAndDepthFlags>",
+   "<isClassFlags>",
+   "<vft>",
+   "<currentThread>",
+   "<recompilationCounter>",
+   "<excp>",
+   "<indexableSize>",
+   "<resolveCheck>",
+   "<arrayTranslate>",
+   "<arrayTranslateAndTest>",
+   "<long2String>",
+   "<bitOpMem>",
+   "<reverseLoad>",
+   "<reverseStore>",
+   "<currentTimeMaxPrecision>",
+   "<headerFlags>",
+   "<singlePrecisionSQRT>",
+   "<countForRecompile>",
+   "<gcrPatchPoint>",
+   "<counterAddress>",
+   "<startPC>",
+   "<compiledMethod>",
+   "<thisRangeExtension>",
+   "<profilingBufferCursor>",
+   "<profilingBufferEnd>",
+   "<profilingBuffer>",
+   "<osrBuffer>",
+   "<osrScratchBuffer>",
+   "<osrFrameIndex>",
+   "<osrReturnAddress>",
+   "<potentialOSRPointHelper>",
+   "<osrFearPointHelper>",
+   "<eaEscapeHelper>",
+   "<instanceShape>",
+   "<instanceDescription>",
+   "<descriptionWordFromPtr>",
+   "<objectEqualityComparison>",
+   "<synchronizedFieldLoad>",
+   "<atomicAdd>",
+   "<atomicFetchAndAdd>",
+   "<atomicFetchAndAdd32Bit>",
+   "<atomicFetchAndAdd64Bit>",
+   "<atomicSwap>",
+   "<atomicSwap32Bit>",
+   "<atomicSwap64Bit>",
+   "<atomicCompareAndSwapReturnStatus>",
+   "<atomicCompareAndSwapReturnValue>",
+   "<jProfileValueSymbol>",
+   "<jProfileValueWithNullCHKSymbol>"
+   };
+
+
+const char *
+OMR::SymbolReferenceTable::getNonHelperSymbolName(CommonNonhelperSymbol nonHelper)
+   {
+   TR_ASSERT_FATAL(nonHelper >= OMRfirstPrintableCommonNonhelperSymbol &&
+                   nonHelper <= OMRlastPrintableCommonNonhelperSymbol,
+                   "unknown non helper %" OMR_PRId32, static_cast<int32_t>(nonHelper));
+
+#if 0
+   static_assert(sizeof(_commonNonHelperSymbolNames)/sizeof(_commonNonHelperSymbolNames[0]) ==
+                 static_cast<int32_t>(OMRlastPrintableCommonNonhelperSymbol - OMRfirstPrintableCommonNonhelperSymbol + 1),
+                 "_commonNonHelperSymbolNames array must match CommonNonHelperSymbol enumeration");
+#endif
```

Comment:
Would it be useful to enable this?

Reply:
Yes.  This is part of a staged commit so that downstream projects can catch up.  It is enabled in this subsequent PR -> https://github.com/eclipse/omr/pull/5822

Sorry, I should have pointed out the fact that these changes are being added incrementally.

Reply:
Thanks, looks good then. I'm happy to merge as soon as testing passes.